src/Microsoft.ML.Transforms/CategoricalStaticExtensions.tt

```diff
@@ -0,0 +1,80 @@
+
+using System;
+using Microsoft.ML.Data.StaticPipe.Runtime;
+
+namespace Microsoft.ML.Runtime.Data
+{
+    public static partial class CategoricalStaticExtensions
+    {
+        // Do not edit this file directly. Rather, it is generated out of TermStaticExtensions.tt.
+<#
+// Let's skip the time-based types for now.
+foreach (string typeName in new string[] { "string", "float", "double", "sbyte", "short", "int", "long", "byte", "ushort", "uint", "ulong", "bool" }) {
+bool startRegionBlock = true;
+#>
+
+        #region For <#=typeName#> inputs.
+<#
+foreach (bool inputIsKey in new bool[] { false, true }) {
+foreach (string arityName in new string[] { "Scalar", "Vector", "VarVector" }) {
+string onFitType = typeName == "string" ? "ReadOnlyMemory<char>" : typeName;
+bool omitInputArity = arityName == "Scalar" && inputIsKey;
+bool isNumeric = typeName != "string" && typeName != "bool";
+bool isScalar = arityName == "Scalar";
+bool isVarVector = arityName == "VarVector";
+
+if (!startRegionBlock) { // Put lines between the declarations to make them look pretty, but not after the region tag.
+#>
+
+<#}
+startRegionBlock = false;
+#>
+        /// <summary>
+        /// Map values to a key-value representation, where the key type's values are those values observed in the input
+        /// during fitting. During transformation, any values unobserved during fitting will map to the missing key.
+<#
+if (typeName == "string") { #>
+        /// Because the empty string is never entered into the dictionary, it will always map to the missing key.
+<# }
+if (typeName == "float" || typeName == "double") { #>
+        /// Because <c>NaN</c> floating point values are never entered into the dictionary, and they will always map to the missing key.
+<# }
+if (!isScalar && isNumeric && !inputIsKey) { #>
+        /// Zero is considered a valid value and so will be entered into the dictionary if observed. The potential perf
+        /// implication in that case is that sparse input numeric vectors will map to dense output key vectors.
+<# }
+if (inputIsKey) { #>
+        /// We are inputting a key type with values, and in that case the dictionary is considered to be built over the
+        /// values of the keys, rather than the keys themselves. This also mean the key-values learned for the output
+        /// will be a subset of the key-values in the input.
+<# }
+
+#>
+        /// </summary>
+        /// <param name="input">The input column.</param>
+        /// <param name="outputKind"></param>
+        /// <param name="order">The ordering policy for what order values will appear in the enumerated set.</param>
+        /// <param name="maxItems">The maximum number of items.</param>
+
+        /// <returns>The key-valued column.</returns>
+        public static <#=isVarVector?"Var":""#>Vector<float> OneHotEncoding<#=inputIsKey?"<T>":""#>(this <#=omitInputArity?"":arityName+"<"#><#=inputIsKey?"Key<T, ":""#><#=typeName#>><#=inputIsKey&&!isScalar?">":""#> input,
```

Comment:
>isVarVector [](start = 25, length = 11)

I need to handle situation in which VarVector<T> becomes Vector<float> if user choose bagging output.
Only possible solution coming to my mind so far is to have OneHotEncodingBagged and OneHotEncoding methods and expose enumeration without bagging option